modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/components/CodeEditor/Sidebar.tsx

```diff
@@ -29,34 +55,26 @@ export function Sidebar({editorRef, elements}: IProps) {
 			<div className="px-3">
 				<h5 className="my-3">{Liferay.Language.get('elements')}</h5>
 
-				{elements.map(({items, label}) => (
-					<Collapsible key={label} label={label}>
-						{items.map((item) => (
-							<Element
-								helpText={item.helpText}
-								key={item.label}
-								label={item.label}
-								onClick={() => handleClick(item)}
-							/>
-						))}
-					</Collapsible>
-				))}
+				{CustomSidebarContent &&
+					CustomSidebarContent({
+						elements,
+						handleElementClick: handleClick,
+					})}
+
+				{!CustomSidebarContent &&
+					elements?.map(({items, label}) => (
+						<Collapsible key={label} label={label}>
+							{items.map((item) => (
+								<Element
+									helpText={item.helpText}
+									key={item.label}
+									label={item.label}
+									onClick={() => handleClick(item)}
+								/>
+							))}
+						</Collapsible>
+					))}
```

Comment:
Hey @naraandrade, I'm wondering if it's really necessary to pass all the elements to the CustomSidebarContent and repeat the code in all the components that will use this prop. I think we can avoid this code repetition:

```tsx
{elements?.map(({items, label}) => (
    <Collapsible key={label} label={label}>
	    {items.map((item) => (
		    <Element
			    helpText={item.helpText}
			    key={item.label}
			    label={item.label}
			    onClick={() => handleElementClick(item)}
		    />
	    ))}
    </Collapsible>
))}
```

I think we can simply pass the ReactNode as CustomSidebarContent, and whoever implements this component will have full control of the sidebarElements, as they will have access to these elements from the parent component. what do you think about?

Reply:
@danvb112 I thought about doing it that way but I came across a limitation. The element's click needs to be handled inside the sidebar, since we need the reference for the editor to control the text that will be added after clicking the element. It wouldn't make sense to export the editorRef so I made the decision of handling the elements that way. The current solution would also support rendering the sidebar elements in any way the dev could want, so it wouldn't necessary be a code repetion, only if they want it rendered the same way.